docs/cpp_build_systems.md

```diff
@@ -0,0 +1,350 @@
+    change.
+*   A `cc_library` is added: the new target may need to be added to the Protobuf
+    `cc_dist_library` targets, as appropriate.
+*   A `cc_library` is deleted: if a `cc_dist_library` depends upon the deleted
+    target, then a build-time error will result. The library needs to be removed
+    from the `cc_dist_library`.
+*   A `cc_test` is added or deleted: test sources are handled by `filegroup`
+    rules defined in the same package as the `cc_test` rule. The `filegroup`s
+    are usually given a name like `"test_srcs"`, and often use `glob()` to find
+    sources. This means that adding or removing a test may not require any extra
+    work, but this can be verified within the same package as the test rule.
+*   Test-only proto files are added: the `proto_library` might need to be added
+    to the file list map in `//pkg:BUILD.bazel`, and then the file added to
+    various build systems. However, most test-only protos are already exposed
+    through libraries like `//src/google/protobuf:test_protos`.
+
+If there are changes, then the regenerated file lists need to be copied back
+into the repo. That way, the corresponding build systems can be used with a git
+checkout, without needing to run Bazel first.
+
+### (Aside) Distribution archives
+
+A very similar set of rules is defined in `//pkg` to build source distribution
+archives for releases. In addition to the full sources, Protobuf releases also
+include source archives sliced by language, so that, for example, a Ruby-based
+project can get just the sources needed to build the Ruby runtime. (The
+per-language slices also include sources needed to build the protobuf compiler,
+so they all effectively include the C++ runtime.)
+
+These archives are defined using rules from the
+[rules_pkg](https://github.com/bazelbuild/rules_pkg) project. Although they are
+similar to `cc_dist_library` and the file list generation rules, the goals are
+different: the build system file lists described above only apply to C++, and
+are organized according to what should or should not be included in different
+parts of the build (e.g., no tests are included in the main library). On the
+other hand, the distribution archives deal with languages other than C++, and
+contain all the files that need to be distributed as part of a release (even for
+C++, this is more than just the C++ sources).
+
+While it might be possible to use information from the `CcFileList` and
+`ProtoFileList` providers to define the distribution files, additional files
+(such as the various `BUILD.bazel` files) are also needed in the distribution
+archive. The lists of distribution files can usually be generated by `glob()`,
+anyhow, so sharing logic with the file list aspects may not be beneficial.
+
+Currently, all of the file lists are checked in. However, it would be possible
+to build the file lists on-the-fly and include them in the distribution
+archives, rather than checking them in to the git repo.
```

Comment:
in to --> into

Reply:
Done (well, dropped some redundant words, anyway).